### ISSUE TITLE
"Mod+Alt+L" does not work as a keybind for ANY action!

### ISSUE THREAD

**Issue by rempas** [YaLTeR/niri#299]
### System Information

* niri version: niri 0.1.5 (6a80078)

* GPU: AMD RX 560

* CPU: AMD Ryzen 5 2400G

### Bug description:

Hello! First of all, thank you for your great work on Niri! I just tried it for the first time, and it looks very cool and things work great except for fractional scaling and the problem I'm going to describe here!

So, the problem is that when I use "Mod+Alt+L" as a keybind, it doesn't work. I have tried it for **set-window-height "-10%";** **set-column-width "-10%";** and **set-column-width "+10%";** and it didn't work in neither. What happens is that, it just doesn't have any effect when I press these keys...

**Comment by YaLTeR** [comment]
Hey, if you're trying in a window then it conflicts with the default Mod+L binding (focus-column-right). It'll work on a TTY.

**Comment by rempas** [comment]
> Hey, if you're trying in a window then it conflicts with the default Mod+L binding (focus-column-right). It'll work on a TTY.

Wow! That's fast! So, I had changed the default keybind and I had actually set "Mod+L" to **focus-column-up** so that's the first thing. Second, this doesn't seem to fix it. If I "free" "Mod+L", "Mod+Alt+L" will still not work...

But in any case, shouldn't these two keybind by unrelated and be allowed to make different actions?

**Comment by YaLTeR** [comment]
So, niri does not currently support binding the same key combo to multiple things. If that happens, it will use the first binding and ignore the rest.

Now, normally this doesn't happen because the config parser will report an error. However, the Mod key is a bit odd because it changes between Super and Alt depending on whether you're running niri on a TTY or as a window. To avoid being annoying, niri will allow bindings that conflict only due to Mod, but it will still use the first binding in case of a conflict.

So in your case, when running as a window, check if you have another Alt+L or Mod+L or Mod+Alt+L binding, since all three will turn into Alt+L (since in a window, Mod = Alt).

**Comment by YaLTeR** [comment]
Oh yeah, there's also an unrelated issue when running as a window: the modifier keys will sometimes get stuck and stop working, which will seem as though the hotkeys aren't working right (to be fixed in Smithay).

Really the windowed mode is in a bit of a rough shape because it's only meant for development and quick testing.

**Comment by rempas** [comment]
Thank you so much for the help! After reading the last two replies, I was able to make it work by uncommenting any "*+L" keybind!

Unfortunately, that makes me workload impossible, but I understand! One question if I may ask, do you plan to allow different combinations to be used in the future?

**Comment by YaLTeR** [comment]
> One question if I may ask, do you plan to allow different combinations to be used in the future?

What exactly do you mean by that? Same key bound to multiple things?

**Comment by rempas** [comment]
> What exactly do you mean by that? Same key bound to multiple things?

I mean, been able to use both "Mod+L" and "Mod+Alt+L" for example. From what I understand you said that this isn't possible now right? And what caused my problem in the first place. Tho now that I'm thinking about it... other keys that had "combos" work without problems so is it "random" where it works and where it doesn't?

**Comment by YaLTeR** [comment]
You can use two binds like this on a TTY where Mod = Super. There the first bind is Super+L and the second is Super+Alt+L so there's no conflict (note there's a default Super+Alt+L bind to swaylock).

**Comment by rempas** [comment]
> You can use two binds like this on a TTY where Mod = Super. There the first bind is Super+L and the second is Super+Alt+L so there's no conflict (note there's a default Super+Alt+L bind to swaylock).

AAAAAAAAAH, I'm an idiot!!!! This is another case of "I thought I tried that" but it seems I didn't, and I did something else instead...

Yeah, commenting out "swaylock" does fix things, finally! I'm very sorry, if I was a little bit more careful, we would have ended this 20 minutes ago, and I wouldn't have wasted your time.

If you allow me, I would like to say that you seem very kind and friendly. Thank you for the help and have a beautiful day! :heart:
